.github/workflows/scheduled.yml

```diff
@@ -15,41 +15,126 @@
 name: "Scheduled Fuzzer Jobs"
 
 on:
+  pull_request:
+    paths:
+      - ".github/workflows/scheduled.yml"
+
   schedule:
     - cron: '0 3 * * *'
 
+  workflow_dispatch:
+    ref:
+      description: 'Ref to checkout out'
+      default: 'main'
+    numThreads:
+      description: 'Number of threads'
+      default: 16
+    maxHighMemJobs:
+      description: 'Number of high memory jobs'
+      default: 8
+    maxLinkJobs:
+      description: 'Maximum number of link jobs'
+      default: 4
+    extraCMakeFlags:
+      description: 'Additional CMake flags'
+      default: ''
+
 defaults:
   run:
     shell: bash
 
+permissions:
+  contents: read
+
 jobs:
-  linux-presto-fuzzer-run:
-    runs-on: 16-core
+  compile:
+    runs-on: 8-core
     timeout-minutes: 120
     env:
       CCACHE_DIR: "${{ github.workspace }}/.ccache/"
       CCACHE_BASEDIR: "${{ github.workspace }}"
       LINUX_DISTRO: "ubuntu"
     steps:
+      - name: "Install mold as default linker"
+        uses: rui314/setup-mold@v1
+
+      - name: "Restore ccache"
+        uses: actions/cache@v3
+        with:
+          path: "${{ env.CCACHE_DIR }}"
+          # We are using the benchmark ccache as it has all 
+          # required features enabled, so no need to create a new one
+          key: ccache-benchmark-${{ github.sha }}
+          restore-keys: |
+            ccache-benchmark-
 
       - name: "Checkout Repo"
         uses: actions/checkout@v3
         with:
+          path: velox
           submodules: 'recursive'
+          ref: "${{ inputs.ref || 'main' }}"
 
       - name: "Install dependencies"
-        run: source ./scripts/setup-ubuntu.sh
+        run: cd  velox && source ./scripts/setup-ubuntu.sh
 
       - name: "Build"
         run: |
-          make debug NUM_THREADS=16 MAX_HIGH_MEM_JOBS=8 MAX_LINK_JOBS=2 EXTRA_CMAKE_FLAGS="-DVELOX_ENABLE_ARROW=ON"
+          cd velox
+          make debug NUM_THREADS="${{ inputs.numThreads || 8 }}" MAX_HIGH_MEM_JOBS="${{ inputs.maxHighMemJobs || 8 }}" MAX_LINK_JOBS="${{ inputs.maxLinkJobs || 4 }}" EXTRA_CMAKE_FLAGS="-DVELOX_ENABLE_ARROW=ON ${{ inputs.extraCMakeFlags }}"
           ccache -s
 
+      - name: Upload presto fuzzer 
+        uses: actions/upload-artifact@v3
+        with:
+          name: presto
+          path: velox/_build/debug/velox/expression/tests/velox_expression_fuzzer_test
+            
+      - name: Upload spark fuzzer 
+        uses: actions/upload-artifact@v3
+        with:
+          name: spark
+          path: |
+            velox/_build/debug/velox/expression/tests/spark_expression_fuzzer_test
```

Comment:
Am I missing something, is the path for spark repeated twice ?

Reply:
One is the aggregation the other the expression fuzzer :)